pkg/commands/root_test.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2023 KStreamer Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package commands
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/spf13/cobra"
+	"gotest.tools/v3/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+// TestRootCmd is a basic test that will be updated later once we have some real functionality to test
+func TestRootCmd(t *testing.T) {
+	// Create a buffer to hold the output
+	var buf bytes.Buffer
+
+	cobra.OnInitialize(
+		func() {
+			// Create a fake clientset
+			clientSet = fake.NewSimpleClientset()
+
+			// Add a couple of namespaces to simulate existing data
+			ns1 := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "test-namespace-1"}}
+			ns2 := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "test-namespace-2"}}
+			clientSet.CoreV1().Namespaces().Create(context.TODO(), ns1, metav1.CreateOptions{})
+			clientSet.CoreV1().Namespaces().Create(context.TODO(), ns2, metav1.CreateOptions{})
+		},
+	)
+
+	// Create a pipe to capture the output
+	r, w, _ := os.Pipe()
+	// Save the original stdout
+	originalOut := os.Stdout
```

Comment:
I think it would be better to replace this with:
```go
rootCmd.SetOut(buf) // no need for a pipe
```
which will redirect output of just the `rootCmd.Run()` call. This requires using `cmd.Printf("Namespace: %s\n", ns.Name)` instead of the `fmt` version in the root.go.

Right now any output like `fmt.Printf()` which is done after the `os.Stdout = w` will not show up on the terminal, so it took me a while to figure out why the tests were exiting with status 1. The issue was missing `KUBECONFIG`, but the information was not displayed as it was output to the pipe and the whole program exited before the pipe was copied to buffer. If you use cobra's printing, then only the output of the command itself will be put to the buffer. Everything from the tests or init commands will be output to the normal Stdout. This makes the code more testable I believe. True, you cannot redirect the output of that init function printing `KUBECONFIG not set.` but the program would exit anyway just after printing this, so there won't be any possible test for it anyway (other than running the CLI as an external process).

Reply:
Thanks for the nice suggestions! Done.